### PR TITLE
feat(training): expose SFT/DPO DCP checkpoints and LR schedules

### PR DESCRIPTION
**Pull request by benjibc**
## Summary

- Add shared LR schedule config and SFT/DPO per-step LR updates, based on the scheduler API shape from https://github.com/fw-ai/cookbook/pull/455. This covers `constant`, `cosine`, `generalized_cosine`, `linear`, and `two_point_linear`, and logs `train/lr`.
- Keep SFT `warmup_steps` as a backwards-compatible alias for `lr_schedule.warmup_steps`.
- Make SFT/DPO periodic DCP saves both resumable and promotable, document the developer flow, and support `cross_job://job_id/checkpoint` checkpoint references.
- Preserve the managed DCP resume path by reusing requested trainer job IDs, registering cleanup for reused jobs, and retrying transient resume-state races.

## Notes

This intentionally wires scheduler support to SFT/DPO first. ORPO/RFT managed wiring is left out for follow-up work.

## Test plan

- `git diff --check`
- `python -m py_compile training/utils/config.py training/utils/__init__.py training/recipes/sft_loop.py training/recipes/dpo_loop.py training/tests/unit/test_lr_schedule_config.py training/tests/unit/test_sft_loop.py training/tests/unit/test_dpo_loop.py`
- Direct scheduler math smoke check via `training/utils/config.py`
- `python -m pytest -q training/tests/unit/test_lr_schedule_config.py training/tests/unit/test_sft_loop.py training/tests/unit/test_dpo_loop.py` is blocked in this local environment because installed `tinker-cookbook==0.2.2` does not export `ParseTermination`; collection fails while importing `training.renderer.mistral`.

---

> [!NOTE]
> **Medium Risk**
> Touches core SFT/DPO training loops (optimizer-step LR updates, step counting, and checkpoint promotion), which can affect training dynamics and resume/promotion behavior if schedule math or total-step estimation is wrong.
> 
> **Overview**
> Adds a shared `LRScheduleConfig` (with validation + `compute_learning_rate`) and wires it into both SFT and DPO so the optimizer LR is recomputed every step, logged as `train/lr`, and configured consistently across recipes (with SFT `warmup_steps` kept as a deprecated alias).
> 
> Changes SFT/DPO periodic DCP checkpointing (`dcp_save_interval`) to save checkpoints as both **resumable and promotable** (training + inference at the same step), and expands `init_from_checkpoint` parsing to accept `cross_job://job_id/checkpoint` references.
> 
> Extends infra provisioning to support a stable `requested_trainer_job_id` reuse/create path and adds retry logic for transient resume-state races when resuming existing trainer jobs; accompanying unit tests and README guidance are updated accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 81e5a8332ab53096587e87c943fd31cadc6e414d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by benjibc** [comment]
Closing: this change should land through the internal cookbook source instead of the public cookbook repo.
